framework/hpp_api_vulkan_sample.cpp

```diff
@@ -0,0 +1,96 @@
+/* Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <hpp_api_vulkan_sample.h>
+
+// Instantiate the default dispatcher
+VULKAN_HPP_DEFAULT_DISPATCH_LOADER_DYNAMIC_STORAGE
+
+bool HPPApiVulkanSample::prepare(vkb::Platform &platform)
+{
+	if (!ApiVulkanSample::prepare(platform))
+	{
+		return false;
+	}
+
+	// initialize function pointers for C++-bindings
+	static vk::DynamicLoader  dl;
+	PFN_vkGetInstanceProcAddr vkGetInstanceProcAddr =
+	    dl.getProcAddress<PFN_vkGetInstanceProcAddr>("vkGetInstanceProcAddr");
+	VULKAN_HPP_DEFAULT_DISPATCHER.init(vkGetInstanceProcAddr);
+	VULKAN_HPP_DEFAULT_DISPATCHER.init(get_instance());
+	VULKAN_HPP_DEFAULT_DISPATCHER.init(get_device().get_handle());
+
+	return true;
+}
+
+vk::CommandBuffer HPPApiVulkanSample::get_command_buffer(size_t index) const
+{
+	return static_cast<vk::CommandBuffer>(draw_cmd_buffers[index]);
+}
+
+size_t HPPApiVulkanSample::get_command_buffers_count() const
+{
+	return draw_cmd_buffers.size();
+}
+
+vkb::core::HPPDevice &HPPApiVulkanSample::get_device() const
+{
+	return static_cast<vkb::core::HPPDevice &>(*device);
```

Comment:
when down casting it's better to use dynamic_cast instead of static_cast.

Reply:
Only if vkb::Device was polymorphic.